### ISSUE TITLE
[a11y]: icon only ghost buttons not visible (again)

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#9934]
### Package

carbon-components

### Browser

Chrome

### Operating System

Windows

### Package version

10.46

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Description

The problem that was fixed in #6748 is back again: The icons in icon-only ghost buttons are not visible in high contrast mode.

Also, the ghost buttons should not have a border.

### WCAG 2.1 Violation

Minimum contrast

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-button--playground

### Steps to reproduce

Set computer to high contrast mode, navigate to https://react.carbondesignsystem.com/?path=/story/components-button--playground, and use the KNOBS to set the button to a GHOST type.

<img width="1108" alt="Screen Shot 2021-10-25 at 11 14 39" src="https://user-images.githubusercontent.com/69599/138625184-503defc5-5245-449d-a612-10fad3994141.png">
.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Yes! #9904 was included in [v10.47.0](https://github.com/carbon-design-system/carbon/releases/tag/v10.47.0)
